src/objective-c/tests/InteropTests.m

```diff
@@ -457,4 +457,99 @@ - (void)testRPCAfterClosingOpenConnections {
   [self waitForExpectationsWithTimeout:TEST_TIMEOUT handler:nil];
 }
 
+- (void)testAlternateDispatchQueue {
+  XCTAssertNotNil(self.class.host);
+  __weak XCTestExpectation *expectation1 = [self expectationWithDescription:@"AlternateDispatchQueue1"];
+
+  NSArray *requests = @[@27182, @8, @1828, @45904];
+  NSArray *responses = @[@31415, @9, @2653, @58979];
+
+  // Set the default dispatch queue
+  NSString *queue1_label = @"test.queue1";
+  NSString *queue2_label = @"test.queue2";
+  dispatch_queue_t queue1 = dispatch_queue_create([queue1_label UTF8String], DISPATCH_QUEUE_SERIAL);
+  dispatch_queue_t queue2 = dispatch_queue_create([queue2_label UTF8String], DISPATCH_QUEUE_SERIAL);
+  [_service setDefaultResponseDispatchQueue:queue1];
+  GRXBufferedPipe *requestsBuffer1 = [[GRXBufferedPipe alloc] init];
+
+  __block int index = 0;
+
+  id request = [RMTStreamingOutputCallRequest messageWithPayloadSize:requests[index]
+                                               requestedResponseSize:responses[index]];
+  [requestsBuffer1 writeValue:request];
+
+  [_service fullDuplexCallWithRequestsWriter:requestsBuffer1
+                                eventHandler:^(BOOL done,
+                                               RMTStreamingOutputCallResponse *response,
+                                               NSError *error) {
+    XCTAssertNil(error, @"Finished with unexpected error: %@", error);
+    XCTAssertTrue(done || response, @"Event handler called without an event.");
+    NSString *label = [NSString stringWithUTF8String:dispatch_queue_get_label(DISPATCH_CURRENT_QUEUE_LABEL)];
+    XCTAssert([label isEqualToString:queue1_label]);
+
+    if (response) {
+      XCTAssertLessThan(index, 4, @"More than 4 responses received.");
```

Comment:
Remove all assertions that aren't related to whether the callback is running on your designated queue, so the test is easy to read.